I'm guessing this is because my test system is F17, but it's not working for me:

<pre>
TASK: [firewalld port=9999/tcp permanent=true state=enabled] ****************** 
failed: [127.0.0.1] => {"failed": true, "parsed": false}
invalid output was: Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-1376430206.26-9017371969185/firewalld", line 1216, in <module>
    main()
  File "/root/.ansible/tmp/ansible-1376430206.26-9017371969185/firewalld", line 190, in main
    if fw.connected == False:
AttributeError: 'FirewallClient' object has no attribute 'connected'
</pre>

Also, before I installed the firewalld package, it was returning ok for the task despite not changing anything, that should probably be a failure.

--- a/library/net_infrastructure/firewalld
+++ b/library/net_infrastructure/firewalld
@@ -59,7 +59,7 @@ options:
     default: 0
 notes:
    - Not tested on any debian based system
-requirements: [ firewalld ]
+requirements: [ firewalld >= 0.2.11 ]
 author: Adam Miller <maxamillion@fedoraproject.org>
 '''
 
@@ -74,6 +74,9 @@ import re
 import sys
 
 try:
+    import firewall.config
+    FW_VERSION = firewall.config.VERSION
+
     from firewall.client import FirewallClient
     fw = FirewallClient()
 except ImportError:
@@ -164,6 +167,10 @@ def main():
         ),
         supports_check_mode=True
     )
+
+    ## Pre-run version checking
+    if FW_VERSION < "0.2.11":
+        module.fail_json(msg='unsupported version of firewalld, requires >= 2.0.11')
 
     ## Global Vars
     changed=False

